aws_lambda_powertools/metrics/metrics.py

```diff
@@ -82,7 +82,12 @@ def clear_metrics(self):
         self.metric_set.clear()
         self.dimension_set.clear()
 
-    def log_metrics(self, lambda_handler: Callable[[Any, Any], Any] = None, capture_cold_start_metric: bool = False):
+    def log_metrics(
+        self,
+        lambda_handler: Callable[[Any, Any], Any] = None,
+        capture_cold_start_metric: bool = False,
+        raise_for_empty_metrics: bool = False,
```

Comment:
This should be an opt-in behaviour unless this is valid for 80% of customers and we have feedback in this area already. 

```suggestion
        raise_on_empty_metrics: bool = True,
```

Reply:
In my opinion, metrics are rarely a part of the core functionality of an application - and 

> I'm trying to think of an actual customer scenario where this would be common to have no validation by default - Could you share one or two use cases where you wouldn't have a metric if you want to use Metrics?
> 
> I thought about scenarios where metric value would be 0, but even then it's a data point rather than missing data point.

In the code I wanted to integrate with, I want to emit a metric when a shopping cart gets updated. My lambda function has several scenarios where the cart doesn't get updated (for example, the product doesn't exist, or something else in the request was invalid). Maybe I _should_ be emitting metrics on those events too, but if my code exits without adding a metric, my expectation is that my function would still execute without throwing an exception.

Reply:
Discussed offline, agreed on warning when metric set is empty.